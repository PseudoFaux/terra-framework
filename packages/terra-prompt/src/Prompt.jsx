import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import Button from 'terra-button';
import { elementType } from 'react-prop-types';
import styles from './Prompt.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /*
  * Title of the prompt.
  */
  title: PropTypes.string,
  /*
  * Message of the prpmpt.
  */
  message: PropTypes.string,
  /* d
  * Array of buttons. Restricted to terra-button.
  */
  actions: PropTypes.arrayOf(elementType(Button)),
};

const defaultProps = {
  title: '',
  message: '',
  onChange: [],
  actions: [],
};

const Prompt = ({ title, message, actions, ...customProps }) => {
  const PromptClassNames = cx([
    'prompt',
    customProps.className,
  ]);

  return (<div {...customProps} className={PromptClassNames}>{title}</div>);
};

Prompt.propTypes = propTypes;
Prompt.defaultProps = defaultProps;

export default Prompt;
