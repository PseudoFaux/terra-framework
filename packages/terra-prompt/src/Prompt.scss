:local {
  .prompt {
    /* TODO: Add styles */
  }
}
