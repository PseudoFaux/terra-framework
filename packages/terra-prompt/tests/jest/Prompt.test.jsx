import React from 'react';
import Prompt from '../../src/Prompt';

describe('Prompt', () => {
  const defaultRender = <Prompt />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.prompt').text()).toEqual('default');
  });

  // Structure Tests
  it('should have the class prompt', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('prompt');
  });
});
