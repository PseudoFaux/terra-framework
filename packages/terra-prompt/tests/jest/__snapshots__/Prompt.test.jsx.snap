// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt should render a default component 1`] = `
<div
  className="prompt"
>
  default
</div>
`;
